bin/init.bash

```diff
@@ -124,391 +124,38 @@ generate_default_config() {
-        none)
-            return
-            ;;
-    esac
-}
-
-# Usage: set_storage_class <config-file>
-# baremetal support is experimental, keep as separate case until stable
-set_storage_class() {
-    file="${1}"
-    if [[ ! -f "${file}" ]]; then
-        log_error "ERROR: invalid file - ${file}"
-        exit 1
-    fi
-    case ${CK8S_CLOUD_PROVIDER} in
-        safespring | citycloud | elastx)
-            storage_class=cinder-csi
-
-            yq4 -i '.networkPolicies.kubeSystem.openstack.enabled = true' "${file}"
-            yq4 -i '.networkPolicies.kubeSystem.openstack.ips = ["set-me"]' "${file}"
-            yq4 -i '.networkPolicies.kubeSystem.openstack.ports = [5000,8774,8776]' "${file}" # Keystone, Nova, Cinder
-            ;;
-
-        upcloud)
-            storage_class=standard
-
-            yq4 -i '.networkPolicies.kubeSystem.upcloud.enabled = true' "${file}"
-            yq4 -i '.networkPolicies.kubeSystem.upcloud.ips = ["94.237.0.0/23"]' "${file}" # api.upcloud.com
-            yq4 -i '.networkPolicies.kubeSystem.upcloud.ports = [443]' "${file}"
-            ;;
-
-        exoscale | baremetal)
-            storage_class=rook-ceph-block
-            ;;
-
-        aws)
-            storage_class=ebs-gp2
-            ;;
-
-        azure)
-            storage_class=standard
-            ;;
-
-        none)
-            return
-            ;;
-    esac
-
-    replace_set_me "${file}" ".storageClasses.default" "\"${storage_class}\""
-}
-
 # Usage: set_object_storage <config-file>
-# baremetal support is experimental, keep as separate case until stable
 set_object_storage() {
     file="${1}"
     if [[ ! -f "${file}" ]]; then
         log_error "ERROR: invalid file - ${file}"
         exit 1
     fi
     case ${CK8S_CLOUD_PROVIDER} in
-        aws | exoscale)
-            object_storage_type="s3"
-            yq4 --inplace '.objectStorage.s3.region = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.regionEndpoint = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.forcePathStyle = false' "${file}"
-            ;;
-
-        citycloud | safespring | upcloud | elastx)
-            object_storage_type="s3"
-            yq4 --inplace '.objectStorage.s3.region = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.regionEndpoint = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.forcePathStyle = true' "${file}"
-            ;;
-
         azure)
-            object_storage_type="azure"
-            yq4 --inplace '.objectStorage.azure.resourceGroup = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.azure.storageAccountName = "set-me"' "${file}"
-            ;;
-
-        baremetal)
-            object_storage_type="none"
+            replace_set_me "${file}" ".objectStorage.azure.resourceGroup" "\"${CK8S_ENVIRONMENT_NAME}-storage\""
+            replace_set_me "${file}" ".objectStorage.azure.storageAccountName" "\"${CK8S_ENVIRONMENT_NAME}\""
```

Comment:
Can this not also be envsubst or something? Would be nice to get rid of every single special case if possible.

Reply:
That would add another special case, also I am not sure but I get the feeling we want to avoid using envsubst :thinking:

What do other people think about this?

Reply:
Yeah true, I was just thinking that the whole point of this restructure was to avoid setting values in init. But I'm not sure what the best approach is for variable values. I know envsubst is not ideal.

Reply:
Since this ends up in the default config, and the environment name is readily available form the environment, then I think it should be fine to do envsubst here as we'd be able to remove the `set_object_storage` entirely.

We could generalise the envsubst we currently do to make it after everything is merged though, instead of just for the base config. 
Then it wouldn't be a special case, rather we'd have the ability to refer to certain dynamic values.